addons/package_hierarchy/models/stock_quant_package.py

```diff
@@ -271,3 +280,36 @@ def _get_package_link_structure(self):
             )
         hierarchy_lines = hierarchy_lines | package_hierarchy_lines
         return hierarchy_lines.ids
+
+    def get_or_construct_hierarchy_from_packages(self):
+        """Getting or Creating (if any of hierarchy nodes doesnt exist) all hierarchy links of
+        ordered packages that are in self recordset.
+
+        :return: recordset of package hierarchy links
+        """
+        # Setting self to sudo to over pass access rights at this moment, have to be reviewed later
+        self = self.sudo()
+        PackageHierarchyLink = self.env["package.hierarchy.link"]
+
+        # Empty recordset
+        package_hierarchies = self.env["package.hierarchy.link"]
+        # Looping in pairs through packages which are ordered
+        for package, parent_package in pairwise(self):
+            hierarchy_link = PackageHierarchyLink.search(
+                [("child_id", "=", package.id), ("parent_id", "=", parent_package.id)], limit=1
+            )
+            if not hierarchy_link:
+                hierarchy_link = PackageHierarchyLink.create({
+                    "child_id": package.id,
+                    "parent_id": parent_package.id
+                })
+            package_hierarchies |= hierarchy_link
+        hierarchy_link = PackageHierarchyLink.search(
+                [("child_id", "=", parent_package.id), ("parent_id", "=", False)], limit=1
```

Comment:
Will parent_package have value if there is only one element in self?

Reply:
Very good spot didnt thought about it :(
Will check , most probably will raise error i think :)